core/src/moves.rs

```diff
@@ -120,6 +121,23 @@ impl Board {
                             })
                         })
                     }));
+
+                    // explode adjacent letters
+                    changes.extend(Direction::iter().flat_map(|direction| {
+                        let square = position.add(*direction);
+                        if let Ok(Square::Occupied(owner, letter)) = self.get(square) {
+                            if owner != player {
+                                bag.return_tile(letter);
+                                return self.clear(square).map(|detail| {
+                                    Change::Board(BoardChange {
+                                        detail,
+                                        action: BoardChangeAction::Exploded,
+                                    })
+                                });
+                            }
+                        }
+                        None
+                    }));
```

Comment:
Could this instead rely on the existing `board.neighbouring_squares(position)`?

https://github.com/TruncateGame/Truncate/blob/c47641ff3efeaf47190a2ad512f4aa385bf1d1be/src/board.rs#L173-L184

Reply:
Yeah, I think it's probably clearer (though about equally verbose).